src/seedsigner/models/decode_qr.py

```diff
@@ -940,61 +939,75 @@ def __init__(self):
+            elif addr_prefix in ["m", "n"]:
+                self.address_type = (SettingsConstants.LEGACY_P2PKH, SettingsConstants.TESTNET)
 
-                elif r == "3":
-                    # Nested segwit single sig (p2sh-p2wpkh), nested segwit multisig (p2sh-p2wsh), or legacy multisig (p2sh); mainnet
-                    # TODO: Would be more correct to use a P2SH constant
-                    self.address_type = (SettingsConstants.NESTED_SEGWIT, SettingsConstants.MAINNET)
+            elif addr_prefix == "3":
+                # Nested segwit single sig (p2sh-p2wpkh), nested segwit multisig (p2sh-p2wsh), or legacy multisig (p2sh); mainnet
+                # TODO: Would be more correct to use a P2SH constant
+                self.address_type = (SettingsConstants.NESTED_SEGWIT, SettingsConstants.MAINNET)
 
-                elif r == "2":
-                    # Nested segwit single sig (p2sh-p2wpkh), nested segwit multisig (p2sh-p2wsh), or legacy multisig (p2sh); testnet / regtest
-                    self.address_type = (SettingsConstants.NESTED_SEGWIT, SettingsConstants.TESTNET)
+            elif addr_prefix == "2":
+                # Nested segwit single sig (p2sh-p2wpkh), nested segwit multisig (p2sh-p2wsh), or legacy multisig (p2sh); testnet / regtest
+                self.address_type = (SettingsConstants.NESTED_SEGWIT, SettingsConstants.TESTNET)
 
-                elif r == "bc1q":
-                    # Native Segwit (single sig or multisig), mainnet 
-                    self.address_type = (SettingsConstants.NATIVE_SEGWIT, SettingsConstants.MAINNET)
+            elif addr_prefix == "bc1q":
+                # Native Segwit (single sig or multisig), mainnet 
+                self.address_type = (SettingsConstants.NATIVE_SEGWIT, SettingsConstants.MAINNET)
 
-                elif r == "tb1q":
-                    # Native Segwit (single sig or multisig), testnet
-                    self.address_type = (SettingsConstants.NATIVE_SEGWIT, SettingsConstants.TESTNET)
+            elif addr_prefix == "tb1q":
+                # Native Segwit (single sig or multisig), testnet
+                self.address_type = (SettingsConstants.NATIVE_SEGWIT, SettingsConstants.TESTNET)
 
-                elif r == "bcrt1q":
-                    # Native Segwit (single sig or multisig), regtest
-                    self.address_type = (SettingsConstants.NATIVE_SEGWIT, SettingsConstants.REGTEST)
+            elif addr_prefix == "bcrt1q":
+                # Native Segwit (single sig or multisig), regtest
+                self.address_type = (SettingsConstants.NATIVE_SEGWIT, SettingsConstants.REGTEST)
 
-                elif r == "bc1p":
-                    # Native Segwit (single sig or multisig), mainnet 
-                    self.address_type = (SettingsConstants.TAPROOT, SettingsConstants.MAINNET)
+            elif addr_prefix == "bc1p":
+                # Native Segwit (single sig or multisig), mainnet 
```

Comment:
This and following 2 "elif" comments could be Taproot instead of Native Segwit

Reply:
Removed those code comments entirely as they don't add any clarifying information.